.github/workflows/test.yml

```diff
@@ -23,17 +19,39 @@ on:
 
 jobs:
   ci:
-    runs-on: <image-name>
+    runs-on: ubuntu-20.04
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
 
-      - name: Use <setup tooling>
-        uses: <action to setup tooling>
+      - name: Setup Odin
+        uses: laytan/setup-odin@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+          # TODO: Pull a specific version of Odin compatible with the OS.
+          # The current setting pulls the latest version of Odin from the repo, and rebuilds it from
+          # scratch. This is slow, and clutters the action's log. Rather ensure that the OS has the
+          # right dependencies for a recent versioned release of the compiler.
+          release: false
+
+          llvm-version: 14
 
       - name: Install project dependencies
-        run: <install dependencies>
+        run: bin/fetch-configlet
+
+      - name: Run the project linter
+        run: bin/configlet lint
 
       - name: Verify all exercises
         run: bin/verify-exercises
+
+      - name: Apply code formatting to all .odin files
```

Comment:
This should probably be a separate GHA since this one is about checking if the examples passes the test suites. The formatter should fail if something is found so the contributor can fix it themselves. Several tracks follow that setup.

Reply:
I'll log code formatting as a separate issue to factor it out of this PR. I've removed all automatic formatting from the current PR.